<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.189">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Dynamic and Reproducible Science in Quarto</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>


<script src="QUARTOREP_files/libs/clipboard/clipboard.min.js"></script>
<script src="QUARTOREP_files/libs/quarto-html/quarto.js"></script>
<script src="QUARTOREP_files/libs/quarto-html/popper.min.js"></script>
<script src="QUARTOREP_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="QUARTOREP_files/libs/quarto-html/anchor.min.js"></script>
<link href="QUARTOREP_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="QUARTOREP_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="QUARTOREP_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="QUARTOREP_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="QUARTOREP_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Dynamic and Reproducible Science in Quarto</h1>
</div>



<div class="quarto-title-meta">

    
    
  </div>
  

</header>

<section id="introduction" class="level1">
<h1>Introduction</h1>
<section id="my-question" class="level4">
<h4 class="anchored" data-anchor-id="my-question"><strong>My question:</strong></h4>
<p>Do medium ground finches (<em>Geospiza fortis</em>) of Daphne Major (Galapagos) undergo natural selection of beak depth in response to food availability, due to drought?</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/Medium%20Ground%20Finch.jpeg" class="img-fluid figure-img" alt="Medium ground finch perching"></p>
<p></p><figcaption class="figure-caption">Medium ground finch perching</figcaption><p></p>
</figure>
</div>
</section>
<section id="hypotheses" class="level4">
<h4 class="anchored" data-anchor-id="hypotheses"><strong>Hypotheses:</strong></h4>
<p>This being an ecological study, my hypotheses are at the population-level. They refer to changes in beak depth <u>across generations</u>.</p>
<p><strong>Null hypothesis (H<sub>0</sub>):</strong> Medium ground finch beak depth <u>does not change</u> after drought stress.</p>
<p><strong>Alternative hypothesis (H<sub>1</sub>):</strong> Medium ground finch beak depth <u>increases</u> after drought stress. <a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a></p>
<p>In order to test these hypotheses, I will conduct a simple linear model analysis of finch beak depth data (measured as the depth at the base of the beak, in mm) in the years before and after a severe drought – occurring in 1977. If the data show a significant increase in beak depth, we can conclude that natural selection has occurred. <span class="citation" data-cites="fowler2013">(<a href="#ref-fowler2013" role="doc-biblioref">Fowler 2013</a>)</span></p>
</section>
</section>
<section id="methods---statistical-analysis" class="level1">
<h1>Methods - Statistical Analysis</h1>
<p>The Grant dataset includes more than 20 generations of Medium ground finch beak depth data. Its format includes 178 observations on the following 2 variables: Year (1976 and 1978), and Beak Depth (mm).</p>
<p>I have chosen to build a <u>simple linear model</u> to assess whether these finches undergo natural selection of beak depth in response to food availability. Simple linear models show the dependence of a regression target (here, Beak Depth) on Year.</p>
<p>The simple linear model assumes that y can be expressed by a weighted sum of its (p) features with an individual error (epsilon):</p>
<p><span class="math inline">\(y = \beta_0 + \beta_1x_1 + ... + \beta_px_p + \epsilon\)</span></p>
<p>In order to analyse the outputs of this simple linear model, I have chosen to conduct <u>linear regression analysis</u>.</p>
<div class="callout-tip callout callout-style-default callout-captioned">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
Simple linear models are easy to interpret, can be annotated with 95% confidence intervals, and their assumptions can be rigorously tested.&nbsp;
</div>
</div>
<div class="callout-body-container callout-body">

</div>
</div>
<p><span class="citation" data-cites="hastie2017">(<a href="#ref-hastie2017" role="doc-biblioref">Hastie 2017</a>)</span></p>
</section>
<section id="results" class="level1">
<h1>Results</h1>
<p>Initially, I built a multi-panel figure in order to investigate the covariation within the raw data, between Beak Depth (mm) and Year.</p>
<p><span class="citation" data-cites="wickham2023">(<a href="#ref-wickham2023" role="doc-biblioref">Wickham 2023</a>)</span></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>renv<span class="sc">::</span><span class="fu">restore</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>- The library is already synchronized with the lockfile.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="do">## load the packages </span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(broom)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gridExtra)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(hrbrthemes)</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ragg)</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rmarkdown)</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SemiPar)</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Sleuth3)</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(svglite)</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(viridis)</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a><span class="do">##assign the data </span></span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>finchData <span class="ot">&lt;-</span> case0201 </span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a><span class="do">## set "Year" as a factor </span></span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>finchData<span class="sc">$</span>Year <span class="ot">&lt;-</span> <span class="fu">factor</span>(finchData<span class="sc">$</span>Year)</span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a><span class="do">## building a multi-panel exploratory plot</span></span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a><span class="co">#start with histogram </span></span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>histogram <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(finchData, <span class="fu">aes</span>(<span class="at">x =</span> Depth, <span class="at">fill =</span> Year)) <span class="sc">+</span> <span class="co">#loading data and setting axis</span></span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth =</span> <span class="dv">1</span>, <span class="at">color =</span> <span class="st">"darkgrey"</span>, <span class="at">alpha =</span> <span class="fl">0.75</span>) <span class="sc">+</span> <span class="co">#histogram aesthetics </span></span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_freqpoly</span>(<span class="fu">aes</span>(<span class="at">color =</span> Year), <span class="at">binwidth =</span> <span class="dv">1</span>, <span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span>  <span class="co">#frequency polygon aesthetics</span></span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis</span>(<span class="at">discrete =</span> <span class="cn">TRUE</span>, <span class="at">alpha =</span> <span class="fl">0.75</span>) <span class="sc">+</span>  <span class="co">#colour-blind friendly palette</span></span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> Year, <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span>  <span class="co">#setting up individual plots for interpretability</span></span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Beak Depth (mm)"</span>, <span class="co">#x-axis + appropriate units </span></span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Frequency"</span>) <span class="sc">+</span> <span class="co">#y-axis (does not require units)</span></span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span> <span class="co">#an appropriate theme for publication-quality figures </span></span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>( <span class="co">#fine-tuning aesthetic elements of the graph, for publication-quality</span></span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"right"</span>, </span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">vjust =</span> <span class="dv">0</span>, <span class="at">size =</span><span class="dv">15</span>, <span class="at">face =</span> <span class="st">"bold"</span>, <span class="at">family =</span> <span class="st">"Times New Roman"</span>),</span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">12</span>, <span class="at">family =</span> <span class="st">"Times New Roman"</span>),</span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">12</span>, <span class="at">family =</span> <span class="st">"Times New Roman"</span>),</span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">9</span>, <span class="at">family =</span> <span class="st">"Times New Roman"</span>),</span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">9</span>, <span class="at">family =</span> <span class="st">"Times New Roman"</span>),</span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.key.size =</span> <span class="fu">unit</span>(<span class="fl">0.75</span>, <span class="st">"cm"</span>),</span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="fl">11.5</span>, <span class="at">family =</span> <span class="st">"Times New Roman"</span>),</span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>, <span class="at">family =</span> <span class="st">"Times New Roman"</span>),</span>
<span id="cb3-45"><a href="#cb3-45" aria-hidden="true" tabindex="-1"></a>    <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>, <span class="at">family =</span> <span class="st">"Times New Roman"</span>)</span>
<span id="cb3-46"><a href="#cb3-46" aria-hidden="true" tabindex="-1"></a>  ) </span>
<span id="cb3-47"><a href="#cb3-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-48"><a href="#cb3-48" aria-hidden="true" tabindex="-1"></a><span class="co">#next, creating the boxplot</span></span>
<span id="cb3-49"><a href="#cb3-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-50"><a href="#cb3-50" aria-hidden="true" tabindex="-1"></a>boxplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(finchData, <span class="fu">aes</span>(<span class="at">x =</span> Year, <span class="at">y =</span> Depth, <span class="at">fill =</span> Year)) <span class="sc">+</span></span>
<span id="cb3-51"><a href="#cb3-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb3-52"><a href="#cb3-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis</span>(<span class="at">discrete =</span> <span class="cn">TRUE</span>, <span class="at">alpha =</span> <span class="fl">0.75</span>) <span class="sc">+</span>  </span>
<span id="cb3-53"><a href="#cb3-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">color =</span> <span class="st">"darkgrey"</span>, <span class="at">size =</span> <span class="dv">2</span>, <span class="at">alpha =</span> <span class="fl">0.75</span>) <span class="sc">+</span> <span class="co">#adding jitter to show raw data, as is appropriate for an exploratory figure </span></span>
<span id="cb3-54"><a href="#cb3-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_ipsum</span>() <span class="sc">+</span> <span class="co">#another appropriate theme for publication-quality figures </span></span>
<span id="cb3-55"><a href="#cb3-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>( <span class="co">#more modifications to enhance readability </span></span>
<span id="cb3-56"><a href="#cb3-56" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"right"</span>,</span>
<span id="cb3-57"><a href="#cb3-57" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">vjust =</span> <span class="fl">7.5</span>, <span class="at">size =</span> <span class="dv">15</span>, <span class="at">face =</span> <span class="st">"bold"</span>, <span class="at">family =</span> <span class="st">"Times New Roman"</span>),</span>
<span id="cb3-58"><a href="#cb3-58" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">12</span>, <span class="at">family =</span> <span class="st">"Times New Roman"</span>),</span>
<span id="cb3-59"><a href="#cb3-59" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">12</span>, <span class="at">family =</span> <span class="st">"Times New Roman"</span>),</span>
<span id="cb3-60"><a href="#cb3-60" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">9</span>, <span class="at">family =</span> <span class="st">"Times New Roman"</span>),</span>
<span id="cb3-61"><a href="#cb3-61" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">9</span>, <span class="at">family =</span> <span class="st">"Times New Roman"</span>),</span>
<span id="cb3-62"><a href="#cb3-62" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.key.size =</span> <span class="fu">unit</span>(<span class="fl">1.5</span>, <span class="st">"cm"</span>),</span>
<span id="cb3-63"><a href="#cb3-63" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="fl">11.5</span>, <span class="at">family =</span> <span class="st">"Times New Roman"</span>),</span>
<span id="cb3-64"><a href="#cb3-64" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>, <span class="at">family =</span> <span class="st">"Times New Roman"</span>)</span>
<span id="cb3-65"><a href="#cb3-65" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb3-66"><a href="#cb3-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Year"</span>) <span class="sc">+</span> <span class="co">#x-axis label (does not require units) </span></span>
<span id="cb3-67"><a href="#cb3-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Depth (mm)"</span>) <span class="co">#y-axis label + appropriate units </span></span>
<span id="cb3-68"><a href="#cb3-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-69"><a href="#cb3-69" aria-hidden="true" tabindex="-1"></a><span class="co">#next, combining both plots into a multi-panel figure </span></span>
<span id="cb3-70"><a href="#cb3-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-71"><a href="#cb3-71" aria-hidden="true" tabindex="-1"></a><span class="fu">grid.arrange</span>(</span>
<span id="cb3-72"><a href="#cb3-72" aria-hidden="true" tabindex="-1"></a>  histogram <span class="sc">+</span> </span>
<span id="cb3-73"><a href="#cb3-73" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">"(Fig 1a) | Frequency Distributions of Beak Depths Before and After Drought"</span>),</span>
<span id="cb3-74"><a href="#cb3-74" aria-hidden="true" tabindex="-1"></a>  boxplot <span class="sc">+</span> </span>
<span id="cb3-75"><a href="#cb3-75" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">"(Fig 1b) | Ranges of Beak Depths Before and After Drought"</span>),</span>
<span id="cb3-76"><a href="#cb3-76" aria-hidden="true" tabindex="-1"></a>  <span class="at">nrow =</span> <span class="dv">2</span></span>
<span id="cb3-77"><a href="#cb3-77" aria-hidden="true" tabindex="-1"></a>) <span class="co">#figure titles enhance the ease of which to reference body text to figures</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="QUARTOREP_files/figure-html/multi-panel%20descriptive-1.png" class="img-fluid figure-img" width="960"></p>
<p></p><figcaption class="figure-caption">Multi-panel figure describing the raw data. 1976 = purple. 1978 = yellow. | Fig 1a shows two histograms. The x-axis represents Beak Depth (measured in mm), and the y-axis represents Frequency for that year. Frequency polygons are illustrated in red (1976) and blue (1978). | Fig 1b shows a box plot. The x-axis represents the Year, and the y-axis represents the Beak Depth (mm). Grey dots illustrate the raw data points.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Fig 1a shows two histograms, which depict the frequency data of beak depths for both 1976 (before drought), and 1979 (after drought). Histograms are useful for illustrating trends in data <span class="citation" data-cites="tufte2001">(<a href="#ref-tufte2001" role="doc-biblioref">Tufte 2001a</a>)</span>. Overlayed onto the histograms are frequency polygons, connecting the midpoints of each bin to illustrate the overall distribution of the data. In 1976, the purple histogram appears to have a right-skewed distribution, when compared to 1978 (the yellow histogram). This is further exemplified by the red frequency polygon for 1976 having a greater right-skew than the blue frequency polygon for 1978. As such, the raw data gives an indication of an increase in Beak Depth, between the two years.</p>
<p>Fig 1b shows a box plot, which depicts the range of values for beak depths in both years. We can see the outliers (dots), the range (whiskers) and Quartiles (Qs) 1, 2 and 3 illustrated by the bounds of the box. Box plots are useful because they are compact, and easy to interpret <span class="citation" data-cites="tufte2001a">(<a href="#ref-tufte2001a" role="doc-biblioref">Tufte 2001b</a>)</span>. Our box plot shows less lower-bounded outliers in 1978, as well as smaller range, and an increase in Qs 1, 2 and 3 between 1976 and 1978. This is further indication of an increase in beak depth, after the 1977 drought event.</p>
<p>For reproducibility, it is important that my figures are saved, for maximum transparency <span class="citation" data-cites="schwab2022">(<a href="#ref-schwab2022" role="doc-biblioref">Schwab 2022a</a>)</span>. They can be found in the “figures” folder of my QUARTOREP Project, and in my GitHub repository, here: <a href="https://github.com/daisyjeans/QUARTOREP" title="click">GitHub Repository</a>. I give the code for my figure-saving function, which itself is also saved in the “functions” folder of my QUARTOREP Project and GitHub Repository, below:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="do">## In order to save my figure, I have created a function, available in the "functions" folder of my Project, and in my GitHub Repository: </span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">here</span>(<span class="st">"functions"</span>, <span class="st">"saving_figures.R"</span>))</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="do">## Then call this function with regard to the exploratory figure:</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="fu">save_figure_png</span>(histogram,</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>                <span class="fu">here</span>(<span class="st">"figures"</span>, <span class="st">"Figure1_histogram.png"</span>), </span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">size =</span> <span class="dv">50</span>, <span class="at">res =</span> <span class="dv">500</span>, <span class="at">scaling =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save_figure_png</span>(boxplot,</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>                <span class="fu">here</span>(<span class="st">"figures"</span>, <span class="st">"Figure1_boxplot.png"</span>), </span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">size =</span> <span class="dv">50</span>, <span class="at">res =</span> <span class="dv">500</span>, <span class="at">scaling =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
</div>
<p>After appreciating the raw data, it is necessary to conduct formal analysis. As such, having built my simple linear model, I conducted a <u>linear regression analysis</u>, with the following output:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>simplemodel <span class="ot">&lt;-</span> <span class="fu">lm</span>(Depth <span class="sc">~</span> Year, <span class="at">data =</span> finchData) <span class="co">#assinging a simple linear regression model of depth against year for the finch dataset</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>slope <span class="ot">&lt;-</span> <span class="fu">signif</span>(simplemodel<span class="sc">$</span>coef[[<span class="dv">2</span>]], <span class="dv">5</span>) <span class="co">#assinging the slope value, for later extraction, for ease of visibility </span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>p_value <span class="ot">&lt;-</span> <span class="fu">signif</span>(<span class="fu">summary</span>(simplemodel)<span class="sc">$</span>coef[<span class="dv">2</span>, <span class="dv">4</span>], <span class="dv">5</span>) <span class="co">#same goes for the p-value</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>r_squared <span class="ot">&lt;-</span> <span class="fu">signif</span>(<span class="fu">summary</span>(simplemodel)<span class="sc">$</span>r.squared, <span class="dv">5</span>) <span class="co">#same goes for R^2 value </span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="co"># next, we create a dataframe to store our outputs of interest</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>model_summary <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">Parameter =</span> <span class="fu">c</span>(<span class="st">"Slope"</span>, <span class="st">"P-value"</span>, <span class="st">"R-squared"</span>),</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">Value =</span> <span class="fu">c</span>(slope, p_value, r_squared)</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a><span class="co">#finally, we use the knitr package to make the outputted table more attractive </span></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(model_summary, <span class="at">caption =</span> <span class="st">"Simple Linear Regression Analysis Summary Table"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<caption>Simple Linear Regression Analysis Summary Table</caption>
<thead>
<tr class="header">
<th style="text-align: left;">Parameter</th>
<th style="text-align: right;">Value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Slope</td>
<td style="text-align: right;">0.6685400</td>
</tr>
<tr class="even">
<td style="text-align: left;">P-value</td>
<td style="text-align: right;">0.0000086</td>
</tr>
<tr class="odd">
<td style="text-align: left;">R-squared</td>
<td style="text-align: right;">0.1066300</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>From this output table of my simple linear model, we can see that Beak Depth increased by 0.6685400mm between 1976 and 1978.</p>
<p>Our H<sub>0</sub> assumes that there is no change in Beak Depth from 1976 to 1978. We would fail to reject H<sub>0</sub> if the inbuilt t-test, which compares Beak Depths between the two years and derives a t-statistic with an associated p-value, was non-significant. However, our t-test returns a highly significant associated p-value of 0.0000086. This is much lower than the traditional alpha threshold of 0.05. As such, we reject H<sub>0</sub>.</p>
<p>Notably however, the R<sup>2</sup> output of this linear regression analysis = 0.1066300. This means that the Year can explain only approximately 10.7% of the variability in Beak Depth. This implies that other factors; be they physiological, ecological, environmental or genetic, are affecting the change in beak depth, too.</p>
<p>To more easily interpret of these findings, I built them into a <u>simple linear regression analysis</u> plot.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>finchData<span class="sc">$</span>Year <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(finchData<span class="sc">$</span>Year) <span class="co">#for linear regression, we must now treat "year" as numerical </span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>resultsfigure <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(finchData, <span class="fu">aes</span>(<span class="at">x =</span> Year, <span class="at">y =</span> Depth)) <span class="sc">+</span> </span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="co">#generates a scatter plot</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">se =</span> <span class="cn">TRUE</span>, <span class="at">color =</span> <span class="st">"purple"</span>) <span class="sc">+</span>  <span class="co">#adding the linear regression with 95% CIs to the scatter plot</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"(Fig 2) | Simple Linear Model of Medium Ground Finch Beak Depth Before + After Drought"</span>,</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Year"</span>,</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Beak Depth (mm)"</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>       ) <span class="sc">+</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>), <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"1976"</span>, <span class="st">"1978"</span>)) <span class="sc">+</span> <span class="co">#defining and labelling the x-values is now necessary, since "year" is being treated as numerical </span></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">1.7</span>, <span class="at">y =</span> <span class="fu">max</span>(finchData<span class="sc">$</span>Depth) <span class="sc">+</span> <span class="fl">0.25</span>,</span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">Slope = "</span>, slope,</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>                         <span class="st">"</span><span class="sc">\n</span><span class="st">P-value = "</span>, p_value), <span class="co">#pasting the most appropriate analysis outputs onto the graph, for ease of perception</span></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>           <span class="at">size =</span> <span class="dv">4</span>, <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">vjust =</span> <span class="fl">0.75</span>) <span class="sc">+</span> </span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span> <span class="co">#another useful palette for publication-quality figures </span></span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>, <span class="at">size =</span> <span class="dv">12</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>)) <span class="co">#adjusting for aesthetics of the title </span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Notably, the confidence intervals (the range for which the Depth estimate meets the “true” Depth value, with 95% confidence) do not overlap. Taken together, the highly statistically significant increase in Beak Depth would indicate that the Medium ground finches have undergone natural selection. Hence, we reject H<sub>0</sub>. <span class="citation" data-cites="svensson2023">(<a href="#ref-svensson2023" role="doc-biblioref">Svensson 2023</a>)</span></p>
<p>It is once again important, in order to ensure maximum reproducibility, that there is a saved copy of my Figure 2 <span class="citation" data-cites="schwab2022a">(<a href="#ref-schwab2022a" role="doc-biblioref">Schwab 2022b</a>)</span>. I save it to the “figures” folder of my QUARTOREP Project using the following code. It can also be found in my <a href="https://github.com/daisyjeans/QUARTOREP" title="click again">GitHub Repository</a>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="do">## I reuse my function to save my results figure: </span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="fu">save_figure_png</span>(resultsfigure,</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>                <span class="fu">here</span>(<span class="st">"figures"</span>, <span class="st">"Figure2_linear_regression.png"</span>), </span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">size =</span> <span class="dv">30</span>, <span class="at">res =</span> <span class="dv">100</span>, <span class="at">scaling =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
</div>
<div class="callout-important callout callout-style-default callout-captioned">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
Simple Linear Models make assumptions that must be verified.
</div>
</div>
<div class="callout-body-container callout-body">

</div>
</div>
<p>To verify that the Grant dataset validates the assumptions of a simple linear model, and concurrent regression analysis, I generated a multi-panel figure of regression diagnostics plots <span class="citation" data-cites="kassambara2018">(<a href="#ref-kassambara2018" role="doc-biblioref">Kassambara 2018</a>)</span>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>)) </span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(simplemodel)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="QUARTOREP_files/figure-html/assumptions-1.png" class="img-fluid figure-img" width="960"></p>
<p></p><figcaption class="figure-caption">A multi-panel figure testing the assumptions of linear regression analysis, with regards to this dataset. The top-left plot shows the Residuals vs Fitted. The top-right plot shows the Normal Q-Q plot. The bottom-left plot shows the Scale-Location plot. The bottom-right plot shows the Residuals vs Leverage plot.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>The top-left plot analyses the Residuals vs the Fitted data. This shows the fitted values (meaning the predicted values under the model) on the x-axis, against the residuals (meaning the difference between the predicted and observed values) on the y-axis. This plot is used to verify that the dataset validates the assumptions of a linear relationship between variables: which is shown by a horizontal line about zero. Here, the red line is horizontal about zero, which demonstrates that there is constantly very little difference between predicted and observed values under the model. As such, this assumption is validated.</p>
<p>The top-right plot shows the Normal Q-Q plot. This shows theoretical quantiles (taken from the normal distribution) on the x-axis, and the standardised residuals (specifically, normalised residuals) of the dataset on the y-axis. It is used to validate the assumption of normal distribution of residuals: which is demonstrated by all the residuals (shown by the circles) aligning to the diagonal dashed line; showing that the actual residuals align with the theoretically normal values. Here, the circles align nicely with the dashed line, thus validating this assumption.</p>
<p>The bottom-left plot shows the Scale-Location plot. This also shows the fitted values on the x-axis, but here against the square root of the standard residuals (for better interpretability) on the y-axis. It is used to validate the assumption of homoscedasticity, or homogeneity of variance: which is indicated by a horizontal line. Here, we do see a horizontal line, showing that there is little deviation in the residuals against the fitted values. This therefore validates this assumption.</p>
<p>Finally, the bottom-right plot shows the Constant Leverage plot. This shows the factor level combinations (here, simply the “Year”, as there is only one factor) on the x-axis, and the standardised residuals on the y-axis. It is used to validate the assumption that the linear model appropriately accounts for the contribution of the Year factor to the Beak Depth result. Here, our red line shows no noticeable curvature and does not deviate significantly from zero; the residuals do not significantly deviate within the factor. This therefore validates this assumption.</p>
<p><span class="citation" data-cites="kassambara2018">(<a href="#ref-kassambara2018" role="doc-biblioref">Kassambara 2018</a>)</span></p>
</section>
<section id="conclusion" class="level1">
<h1>Conclusion</h1>
<p>To conclude, the Grant dataset provides an important opportunity for the study of evolution, by the mechanism of natural selection.</p>
<p>Simple regression analysis reveals that medium ground finches (<em>Geospiza fortis</em>) of Daphne Major (Galapagos) experience increased beak depth in response to food (un)availability, due to drought. Despite linear regression only showing a correlation, and not providing conclusive proof of causation, we have strong statistical evidence to reject H<sub>0</sub>. Corroborating data could be sourced from longitudinal studies or historical drought data, that assess changes in phenotype arising from drought conditions. Causative evidence would require intervention studies that simulated drought conditions for a treatment group and measured their phenotype across generations - although this might raise ethical challenges.</p>
<p>This study is important because it hints at the selective forces which can enforce evolutionary change: here, drought. It is essential to consider both the potential incidence, and the nature, of the evolutionary effects that anthropogenic environmental changes (such as more frequent droughts) might impose upon species in our current (proposed) Anthropocene.</p>
<p>Further studies elucidating the genomics that underpin such evolutionary changes, such as that conducted by <span class="citation" data-cites="sangeet2016">(<a href="#ref-sangeet2016" role="doc-biblioref">Sangeet 2016</a>)</span> are promising. Discovering the linked loci that have undergone natural selection, producing adaptive phenotypes (such as the <em>HMGA2</em> gene in this case), will facilitate greater understanding of the proximate and ultimate causes of evolution by natural selection. Only with the appropriate foresight to predict these changes, facilitated by research efforts such as the Grants’, can we ensure species conservation.</p>

</section>


<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-fowler2013" class="csl-entry" role="doc-biblioentry">
Fowler, Samantha. 2013. <em>Concepts of Biology</em>. 1st ed. Vol. 1. 1st Series 1. Houston, Texas: OpenStax. <a href="https://openstax.org/details/books/concepts-biology">https://openstax.org/details/books/concepts-biology</a>.
</div>
<div id="ref-hastie2017" class="csl-entry" role="doc-biblioentry">
Hastie, Trevor. 2017. <em>The Elements of Statistical Learning: Data Mining, Inference, and Prediction</em>. 2nd ed. Vol. 1. 1st Series 1. Stanford University: Springer Series in Statistics. <a href="https://esl.hohoweiya.xyz/book/The%20Elements%20of%20Statistical%20Learning.pdf">https://esl.hohoweiya.xyz/book/The%20Elements%20of%20Statistical%20Learning.pdf</a>.
</div>
<div id="ref-kassambara2018" class="csl-entry" role="doc-biblioentry">
Kassambara, Alboukadel. 2018. <em>Machine Learning Essentials: Practical Guide in r</em>. 1st ed. Vol. 1. 1st Series 1. Marseille: CreateSpace Independent Publishing Platform. <a href="https://www.sthda.com/english/articles/39-regression-model-diagnostics/161-linear-regression-assumptions-and-diagnostics-in-r-essentials/">https://www.sthda.com/english/articles/39-regression-model-diagnostics/161-linear-regression-assumptions-and-diagnostics-in-r-essentials/</a>.
</div>
<div id="ref-sangeet2016" class="csl-entry" role="doc-biblioentry">
Sangeet, Lamichhaney. 2016. <span>“A Beak Size Locus in Darwin<span>’</span>s Finches Facilitated Character Displacement During a Drought.”</span> <em>Science</em>, Science, 352 (1): 470–74. <a href="https://doi.org/DOI:10.1126/science.aad8786">https://doi.org/DOI:10.1126/science.aad8786</a>.
</div>
<div id="ref-schwab2022" class="csl-entry" role="doc-biblioentry">
Schwab, S. 2022a. <span>“Ten Simple Rules for Good Research Practice.”</span> <em>PLOS Computational Biology</em>, Research practice, 18 (6): e1010139. <a href="https://doi.org/doi.org/10.1371">https://doi.org/doi.org/10.1371</a>.
</div>
<div id="ref-schwab2022a" class="csl-entry" role="doc-biblioentry">
———. 2022b. <span>“Ten Simple Rules for Good Research Practice.”</span> <em>PLOS Computational Biology</em>, Research practice, 18 (6): e1010139. <a href="https://doi.org/doi.org/10.1371">https://doi.org/doi.org/10.1371</a>.
</div>
<div id="ref-svensson2023" class="csl-entry" role="doc-biblioentry">
Svensson, Eric. 2023. <span>“Phenotypic Selection in Natural Populations: What Have We Learned in 40 Years?”</span> <em>Oxford Academic</em>, International journal of organic evolution, 77 (7): 14931504. <a href="https://doi.org/10.1093">https://doi.org/10.1093</a>.
</div>
<div id="ref-tufte2001" class="csl-entry" role="doc-biblioentry">
Tufte, Edward. 2001a. <em>The Visual Display of Quantitative Information</em>. 2nd ed. Vol. 1. 1st Series 1. Yale: Graphics Press. <a href="https://www.econ.upf.edu/~michael/visualdata/tufte-aesthetics_and_technique.pdf">https://www.econ.upf.edu/~michael/visualdata/tufte-aesthetics_and_technique.pdf</a>.
</div>
<div id="ref-tufte2001a" class="csl-entry" role="doc-biblioentry">
———. 2001b. <em>The Visual Display of Quantitative Information</em>. 2nd ed. Vol. 1. 1st Series 1. Yale: Graphics Press. <a href="https://www.econ.upf.edu/~michael/visualdata/tufte-aesthetics_and_technique.pdf">https://www.econ.upf.edu/~michael/visualdata/tufte-aesthetics_and_technique.pdf</a>.
</div>
<div id="ref-wickham2023" class="csl-entry" role="doc-biblioentry">
Wickham, Hadley. 2023. <em>R for Data Science: Import, Tidy, Transform, Visualise and Model Data</em>. 2nd ed. Vol. 1. 1st Series 2. Houston, Texas: O’Reilly Media. <a href="https://r4ds.hadley.nz">https://r4ds.hadley.nz</a>.
</div>
</div></section><section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes"><h2 class="anchored quarto-appendix-heading">Footnotes</h2>

<ol>
<li id="fn1"><p>A one-sided hypothesis was chosen because, hypothetically, finches with <u>larger</u> beaks could open the only drought-surviving seeds, which would be highly tough. This would make them more likely to survive, and pass on this advantageous trait to their offspring.<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
</section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>